packages/compass-preferences-model/src/storage.ts

```diff
@@ -1,39 +1,33 @@
-import { UUID } from 'bson';
-import { storedUserPreferencesProps } from './preferences-schema';
 import { z } from 'zod';
+import { UUID } from 'bson';
 import { UserData } from '@mongodb-js/compass-user-data';
 
-type PreferencesValidator = ReturnType<typeof getPreferencesValidator>;
-export type StoredPreferences = z.output<PreferencesValidator>;
-
-export const getDefaultPreferences = (): StoredPreferences => {
-  return Object.fromEntries(
-    Object.entries(storedUserPreferencesProps)
-      .map(([key, value]) => [key, value.validator.parse(undefined)])
-      .filter(([, value]) => value !== undefined)
-  );
-};
-
-const getPreferencesValidator = () => {
-  const preferencesPropsValidator = Object.fromEntries(
-    Object.entries(storedUserPreferencesProps).map(([key, { validator }]) => [
-      key,
-      validator,
-    ])
-  ) as {
-    [K in keyof typeof storedUserPreferencesProps]: typeof storedUserPreferencesProps[K]['validator'];
-  };
-
-  return z.object(preferencesPropsValidator);
-};
+import {
+  getDefaultsForStoredPreferences,
+  getInitialValuesForStoredPreferences,
+  getPreferencesValidator,
+} from './preferences-schema';
+import type {
+  AllPreferences,
+  StoredPreferences,
+  StoredPreferencesValidator,
+} from './preferences-schema';
+
 export interface BasePreferencesStorage {
   setup(): Promise<void>;
   getPreferences(): StoredPreferences;
   updatePreferences(attributes: Partial<StoredPreferences>): Promise<void>;
 }
 
-export class SandboxPreferences implements BasePreferencesStorage {
-  private preferences = getDefaultPreferences();
+export class InMemoryStorage implements BasePreferencesStorage {
+  private preferences = getInitialValuesForStoredPreferences();
```

Comment:
Would like to point out that this initialisation earlier happened with the defaults for `StoredPreferences`. But now we initialise the preferences with the defaults merged with derived values for `StoredPreferences`.

This change affects the PreferencesAccess.CreateSandbox but I feel that this should still be fine. Please let me know if I overlooked something.

Reply:
@himanshusinghs Did you want to update this based on our conversation today? Or was this something for the next/follow-up PR?